From 0ab007b31713006656367c09a555c73fa1462ca6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:54 -0400
Subject: [tools] perf: Fix memory leak when synthesizing thread records

Message-id: <1410181752-25631-110-git-send-email-jolsa@redhat.com>
Patchwork-id: 91829
O-Subject: [PATCH RHEL7.1 BZ1134356 109/307] perf tools: Fix memory leak when synthesizing thread records
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 574799bfdbc3a13e0b5e2f6af34b761bde743a9a
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Mar 17 10:45:49 2014 +0900

description
===========
Checking default guest machine should be done before allocating event
structures otherwise it'll leak memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index ebb48a6..9d12aa6 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -446,6 +446,9 @@ int perf_event__synthesize_threads(struct perf_tool *tool,
  union perf_event *comm_event, *mmap_event, *fork_event;
  int err = -1;
 
+ if (machine__is_default_guest(machine))
+  return 0;
+
  comm_event = malloc(sizeof(comm_event->comm) + machine->id_hdr_size);
  if (comm_event == NULL)
   goto out;
@@ -458,9 +461,6 @@ int perf_event__synthesize_threads(struct perf_tool *tool,
  if (fork_event == NULL)
   goto out_free_mmap;
 
- if (machine__is_default_guest(machine))
-  return 0;
-
  snprintf(proc_path, sizeof(proc_path), "%s/proc", machine->root_dir);
  proc = opendir(proc_path);
 
-- 
1.7.1